components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultLogoutRequestHandler.java

```diff
@@ -123,6 +125,8 @@ public void handle(HttpServletRequest request, HttpServletResponse response, Aut
             if (authenticatedUserObj instanceof AuthenticatedUser) {
                 authenticatedUser = (AuthenticatedUser) authenticatedUserObj;
             }
+            // Setting the user's organization id to the request to get the org id to log out the user.
+            setUserOrgIdInRequest(authenticatedUser, request);
```

Comment:
can we store the user object in the authentication context and use it? This way we can generalize.

Reply:
We can add the authenticated user object to the request but then the org id extraction part should be done at the authenticator level. When the request comes to the authenticator level, it feels it is better to have the exact attribute to process. WDYT?

Reply:
the framework doesn't need to process a specific attribute. We should handle that specific logic in the authenticators.